### DISCUSSION TITLE
Idea: ページ毎に編集権限範囲が設定できるようにする

### DISCUSSION THREAD

**Discussion by sakazuki** [growilabs/growi#8250] [Ideas / Requests] [answered]
現在v6.2.3の仕様ではログインユーザーは閲覧できるページは編集できますが、特定ページについては指定したメンバー以外に編集させたくないケースがあります。

現状の機能範囲では実現できなさそうですので、これを実現するために、
ページ作成時の閲覧権限範囲とは別に、編集権限範囲を追加するというアイディアはどうでしょうか。

ユースケースとしては

人事グループが作成したお知らせページを、全体グループに対して公開する。
人事グループメンバーだけがお知らせページを編集できる。

などがあります。

**Comment by yuki-takei** [comment, marked as answer]
要望ありがとうございます。

権限設定系のリクエストや議論はこれまでいろいろあったのですが、ご要望いただいているものはこれまであまり聞かなかったため、現時点では需要はそれほどないのではないかという感想を持っています。(とはいえ GROWI.cloud を導入検討されている組織で直近でまったく同じリクエストがあったのですが)


## 挙げていただいたユースケースについて

> ユースケースとしては
> 
> 人事グループが作成したお知らせページを、全体グループに対して公開する。
> 人事グループメンバーだけがお知らせページを編集できる。
> 
> などがあります。

当該機能を実装したとして、あるユーザーグループに対する制限の追加でしかなく「Wiki としてできること」が増えているわけではないので、現時点まであまり需要がなかったのかなと思っています。

現在でも例えば利用人数が数百人～千人くらいの GROWI ではおそらくトップページ `/` などは誰彼構わず編集されると困るのではないかと思いますが、自治で成り立っているのが実情ではないかと思います。つまり Wiki 文化が浸透すれば解決可能なケースであって、仮に浸透しないことを前提とするならその組織では Wiki ではないナレッジベースを採用する必要があります。

## GROWI は Wiki か？
以上の議論を進めていくと、GROWI は Wiki 準拠でいいのか？Wiki 以上の機能を期待される場合実装していくべきか？というアイデンティティの問題になってきます。

Wiki システムは自治が成り立つことを前提に複雑な権限設定系の機構を省きあるいは簡略化することができるはずですが、GROWI はすでにそこそこ複雑な機構を持っています。それでも更にきめ細かな権限設定を欲する管理者・組織にとっては不十分でしょうし、どこまで実装するかは悩ましいところです(権限周りの内部のコードがイケてなくて改修コストが高くなってしまっているところは自覚もありますし同意します)。

要するに検討はもっと必要なんだろうとは思っていて、
検討をそこそこに「とりあえず需要がありそうな機能は片っ端から実装していこう」とするには開発リソースが足りないというのがつらいところです。コミュニティ手動で設計・実装・リファクタが進むようであれば歓迎します。

**Reply by sakazuki** [reply]
レスありがとうございます。
レスを読んで、確かに本アイディアはWikiの哲学や思想というよりは、GROWIをCMS的なツールととらえてのものだなと思いました。

そのうえで、現状のGROWIの権限設定周りが整えばCMSツールとしてもユニークなアプリになりそうな可能性も感じました。
権限設定周り（認可の部分）の検討深化については、議論のベースとなるものを提示できないか少し考えてみます。

その上で改めてチケットを起票しますので、本チケットはいったんクローズとしていただければと思います。
